### PR TITLE
Update check should be async

### PR DESCRIPTION
**Pull request by gabriel**
The updater is single flighted so we want the ticker to be async. If an
old update is blocked (a prompt is up and you are away from computer
for a long time), this allows the update check to keep occurring.

This fixes the issue where coyne came into work on Monday and still had
an old update prompt.

### CODE REVIEW

**Comment by gabriel** [comment]
@cjb @patrickxb 

BTW, I am gonna write a test for this in the morning... I had one for singleflighting but not the actual ticker

**Comment by chrisnojima** [comment]
@cjb @patrickxb bump

File: go/updater/update_checker.go [hunk 1]
```diff
@@ -32,7 +42,7 @@ func (u *UpdateChecker) Check(force bool, requested bool) error {
 	if !requested && !force {
 		if lastCheckedPTime := u.updater.config.GetUpdateLastChecked(); lastCheckedPTime > 0 {
 			lastChecked := keybase1.FromTime(lastCheckedPTime)
-			if time.Now().Before(lastChecked.Add(checkDuration())) {
+			if time.Now().Before(lastChecked.Add(u.checkDuration)) {
```
Comment (patrickxb) [inline go/updater/update_checker.go]: `if time.Since(lastChecked) > u.checkDuration`

File: go/updater/update_checker.go [hunk 2]
```diff
@@ -50,37 +61,46 @@ func (u *UpdateChecker) Check(force bool, requested bool) error {
...
-				u.log.Errorf("Error in update: %s", err)
-			}
+		for _ = range u.ticker.C {
+			go func() {
+				u.log.Debug("Checking for update (ticker)")
+				err := u.Check(false, false)
+				if err != nil {
+					u.log.Errorf("Error in update: %s", err)
+				}
+			}()
```
Comment (patrickxb) [inline go/updater/update_checker.go]: Do you want to make sure that all these goroutines finish before the application exits?  Or does it not matter?

**Comment by patrickxb** [comment]
LGTM pending goroutine cleanup comment

File: go/updater/update_checker.go (hunk 2 above)
Comment (gabriel) [inline go/updater/update_checker.go]: The update is atomic so it can be interrupted at any moment... I might add a todo to add a test for this though

**Comment by gabriel** [comment]
@MarcoPolo and I pair programmed the remote window fix above

**Pull request merged**